classes/rust.bbclass

```diff
@@ -86,6 +86,21 @@ HOST_CPPFLAGS ?= "${CPPFLAGS}"
 
 EXTRA_OECONF_remove = "--disable-static"
 
+do_rust_bin_fixups() {
```

Comment:
The observed behavior is that the compiler inserts a runpath when appropriate but it's a relative path from the workdir to the sysroot. I haven't personally looked at the underlying logic that chooses the path.

It's not pretty, but for now we just override it for any ELFs for which the compiler set a runpath.

Before:

`Library runpath: [$ORIGIN/../../../../../sysroots/qemux86/usr/lib/rust:$ORIGIN/../../../../../sysroots/qemux86/usr/lib/i586-poky-linux/rustlib/i586-poky-linux/lib:/<ABSOLUTE PATH TO TOPDIR>/tmp/sysroots/x86_64-linux/usr/lib/rustlib/i586-poky-linux/lib]`

After patchelf:

`Library runpath: [$ORIGIN:/usr/lib/rust:/usr/lib/i586-poky-linux/rustlib/i586-poky-linux/lib]`

And stripping the .note.rustc is done because the section is huge and not needed on the target, but doesn't get pulled out by strip by default.